environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentStatusUpdateService.java

```diff
@@ -0,0 +1,55 @@
+package com.sequenceiq.environment.environment.service;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.event.Payload;
+import com.sequenceiq.cloudbreak.event.ResourceEvent;
+import com.sequenceiq.environment.api.v1.environment.model.response.SimpleEnvironmentResponse;
+import com.sequenceiq.environment.environment.EnvironmentStatus;
+import com.sequenceiq.environment.environment.domain.Environment;
+import com.sequenceiq.environment.environment.dto.EnvironmentDto;
+import com.sequenceiq.environment.environment.v1.EnvironmentApiConverter;
+import com.sequenceiq.flow.core.CommonContext;
+import com.sequenceiq.notification.NotificationService;
+
+@Service
+public class EnvironmentStatusUpdateService {
```

Comment:
I miss the unit test for this class

Reply:
If I try to simplify this method, I will make it unreadable only. The first 2 lines the same only.